backend/src/test/controller/PromotionController.test.ts

```diff
@@ -0,0 +1,339 @@
+import { getCustomRepository } from 'typeorm';
+import { User } from '../../main/entity/User';
+import { UserRepository } from '../../main/repository/UserRepository';
+import connection from '../repository/BaseRepositoryTest';
+import { Express } from 'express';
+import request from 'supertest';
+import { UserFactory } from '../factory/UserFactory';
+import { registerTestApplication } from './BaseController';
+import { PromotionFactory } from '../factory/PromotionFactory';
+import { DiscountFactory } from '../factory/DiscountFactory';
+import { ScheduleFactory } from '../factory/ScheduleFactory';
+import { PromotionRepository } from '../../main/repository/PromotionRepository';
+import { DiscountType } from '../../main/data/DiscountType';
+import { Promotion } from '../../main/entity/Promotion';
+
+describe('Unit tests for PromotionController', function () {
+  let userRepository: UserRepository;
+  let promotionRepository: PromotionRepository;
+  let app: Express;
+
+  beforeAll(async () => {
+    await connection.create();
+    app = registerTestApplication();
+  });
+
+  afterAll(async () => {
+    await connection.close();
+  });
+
+  beforeEach(async () => {
+    await connection.clear();
+    userRepository = getCustomRepository(UserRepository);
+    promotionRepository = getCustomRepository(PromotionRepository);
+  });
+
+  test('GET /promotions', async (done) => {
+    const user: User = new UserFactory().generate();
+    const discount = new DiscountFactory().generate();
+    const schedule = new ScheduleFactory().generate();
+    const promotion = new PromotionFactory().generate(user, discount, [
+      schedule,
+    ]);
+
+    await userRepository.save(user);
+    await promotionRepository.save(promotion);
+
+    request(app)
+      .get('/promotions')
+      .expect(200)
+      .end((err, res) => {
+        if (err) return done(err);
```

Comment:
Does this statement `done(err)` fail the test if executed?

Reply:
Yup!